openedx_events/apps.py

```diff
@@ -11,3 +33,55 @@ class OpenedxEventsConfig(AppConfig):
     """
 
     name = 'openedx_events'
+
+    def _ensure_signal_config_format(self, event_type, configurations):
+        """
+        Validate signal configuration format.
+
+        Raises:
+            ProducerConfigurationError: If configuration is not valid.
+        """
+        if type(configurations) not in (list, tuple):
+            raise ProducerConfigurationError(
+                event_type=event_type,
+                message="Configuration for event_types should be list or tuple of dictionaries"
+            )
+        for configuration in configurations:
+            if not isinstance(configuration, dict):
+                raise ProducerConfigurationError(
+                    event_type=event_type,
+                    message="One of the configuration object is not a dictionary"
+                )
+            expected_keys = {"topic": str, "event_key_field": str, "enabled": bool}
+            for expected_key, expected_type in expected_keys.items():
+                if expected_key not in configuration:
+                    raise ProducerConfigurationError(
+                        event_type=event_type,
+                        message=f"One of the configuration object is missing '{expected_key}' key."
+                    )
+                if not isinstance(configuration[expected_key], expected_type):
+                    raise ProducerConfigurationError(
+                        event_type=event_type,
+                        message=(f"Expected type: {expected_type} for '{expected_key}', "
+                                 f"found: {type(configuration[expected_key])}")
+                    )
```

Comment:
@tecoholic Since we already have a dependency on `attrs`, we can also use it but it will require some custom code to initialize attrs object from a nested dictionary. Pydantic is better but I am not so sure about adding it as a dependency. As this schema was not that difficult to validate, I went ahead with writing the validator. Lets see what suggestions others have.

Reply:
I don't feel strongly about manual verification versus using attrs. Either version requires some custom code.